nixos/modules/installer/netboot/netboot.nix

```diff
@@ -29,6 +29,22 @@ with lib;
           then []
           else [ pkgs.grub2 pkgs.syslinux ]);
 
+    # By default, using netboot images in virtualized contexts
+    # should not create any disk image ideally, except if
+    # asked explicitly.
+    virtualisation.diskImage = mkDefault null;
+    # We do not want to mount the host Nix store in those situations.
+    virtualisation.mountHostNixStore = mkDefault false;
+    # We need a Nix store image because the squashfs is not present
+    # if you perform a direct boot, for example.
+    virtualisation.useNixStoreImage = mkDefault true;
+    # Though, we still want a writable store through .rw-store
+    virtualisation.writableStore = mkDefault true;
+    # Ideally, we might not want to test the network / firmware.
+    virtualisation.directBoot = mkDefault true;
```

Comment:
```
% nix-build -A biosNetboot nixos/tests/boot.nix
error: The option `virtualisation.directBoot' does not exist. Definition values:
       - In `/home/qyliss/src/nixpkgs/nixos/modules/installer/netboot/netboot.nix':
           {
             _type = "override";
             content = true;
             priority = 1000;
           }
(use '--show-trace' to show detailed location information)
```

We can't set virtualisation options in modules like this, can we, because they're not defined? Or did that change?

Reply:
You're right, I guarded this behind a proper check that those options actually exist as done in test-instrumentation.